### ISSUE TITLE
[0.4.6] Unexpected token: name (urlParts)

### ISSUE THREAD

**Issue by aronlmin** [webpack-contrib/uglifyjs-webpack-plugin#155]
I stepped away from a working project for 4 months, after stepping back into my project, I went to install a js library and had to upgrade my node version in order to get it to work, unfortunately, it broke UglifyJs. I keep getting the following compile error.
```{r, engine='bash', count_lines}
aronlilland:backbone_cac_client (master)$ yarn start
yarn run v1.2.1
$ webpack-dev-server
Project is running at http://localhost:8080/
webpack output is served from /
404s will fallback to /index.html
Hash: aa85d44db256ce65b2bf
Version: webpack 2.7.0
Time: 6887ms
             Asset       Size  Chunks                    Chunk Names
img/img-7e92fd.svg    1.02 kB          [emitted]         
         bundle.js    3.55 MB       0  [emitted]  [big]  main
        index.html  872 bytes          [emitted]         
chunk    {0} bundle.js (main) 3.3 MB [entry] [rendered]
    [0] ./~/react/react.js 56 bytes {0} [built]
  [402] ./src/index.js 2.77 kB {0} [built]
  [403] (webpack)-dev-server/client?http://localhost:8080 7.95 kB {0} [built]
  [426] ./src/app/App.jsx 2.96 kB {0} [built]
  [482] ./src/reducers/Index.jsx 1.31 kB {0} [built]
  [730] ./~/loglevel/lib/loglevel.js 7.74 kB {0} [built]
  [979] ./~/react-tap-event-plugin/src/injectTapEventPlugin.js 1.04 kB {0} [built]
 [1083] ./~/redux-thunk/lib/index.js 529 bytes {0} [built]
 [1088] ./~/strip-ansi/index.js 161 bytes {0} [built]
 [1100] ./~/url/url.js 23.3 kB {0} [built]
 [1102] (webpack)-dev-server/client/overlay.js 3.73 kB {0} [built]
 [1103] (webpack)-dev-server/client/socket.js 1.05 kB {0} [built]
 [1105] (webpack)/hot nonrecursive ^\.\/log$ 160 bytes {0} [built]
 [1106] (webpack)/hot/emitter.js 77 bytes {0} [built]
 [1107] multi (webpack)-dev-server/client?http://localhost:8080 ./src/index.js 40 bytes {0} [built]
     + 1093 hidden modules

ERROR in bundle.js from UglifyJs
Unexpected token: name (urlParts) [bundle.js:35511,4]
Child html-webpack-plugin for "index.html":
    chunk    {0} index.html 542 kB [entry] [rendered]
        [0] ./~/lodash/lodash.js 540 kB {0} [built]
        [1] ./~/html-webpack-plugin/lib/loader.js!./src/index.html 1.21 kB {0} [built]
        [2] (webpack)/buildin/global.js 509 bytes {0} [built]
        [3] (webpack)/buildin/module.js 517 bytes {0} [built]
webpack: Failed to compile.

```

## 1. Check the version of package you are using. If it's not the newest version, update and try again (see changelog while updating!).

uglifyjs-webpack-plugin ^0.4.6
webpack ^2.5.0
unfortunately, if I upgrade to the latest webpack, my entire project breaks.

## 2. If the issue is still there, write a minimal project showing the problem and expected output.
Below is my webpack file, the issue goes away if I disable Uglify
```javascript
const path = require('path')

const HtmlWebpackPlugin = require('html-webpack-plugin')
const ExtractTextPlugin = require('extract-text-webpack-plugin')
const UglifyJSPlugin = require('uglifyjs-webpack-plugin')

const HtmlWebpackPluginConfig = new HtmlWebpackPlugin({
  template: './src/index.html',
  filename: 'index.html',
  inject: 'body'
})

module.exports = {
  context: __dirname,
  entry: './src/index.js',
  output: {
    path: path.resolve('dist'),
    filename: 'bundle.js',
    publicPath: '/'
  },
  externals: {
    'jquery': 'jQuery'
  },
  stats: {
    colors: true,
    reasons: true,
    chunks: false
  },
  module: {
    loaders: [
      { test: /\.js$/, loader: 'babel-loader', exclude: /node_modules/ },
      { test: /\.jsx$/, loader: 'babel-loader', exclude: /node_modules/ },
      { test: /\.css$/, loader: 'style-loader', exclude: /node_modules/ },
      { test: /\.css$/, loader: 'css-loader' },
      { test: /\.(png|jpg|gif|svg)$/, loader: 'file-loader?name=img/img-[hash:6].[ext]' }
    ]
  },
  devServer: {
    historyApiFallback: true
  },
  plugins: [
    HtmlWebpackPluginConfig,
    new UglifyJSPlugin({mangle: false, minimize: false, compress: false}),
    new ExtractTextPlugin({ filename: 'css/style.css', disable: false, allChunks: true })
  ]
}
```
## 3. Link to the project and mention Node version and OS in your report.
repo is a private repo.
using node v8.8.1
MacOS High Sierra version 10.13
**IMPORTANT! You should use [Stack Overflow](https://stackoverflow.com/) for support related questions.**

**Comment by gabefinch** [comment]
I'm experienceing an identical compile error on a Windows 10 machine. My package versions: `
    "uglifyjs-webpack-plugin": "^1.0.1",
    "webpack": "^3.8.1",
    "webpack-dev-server": "^2.9.3",
`
To the best of my knowledge I am specifying the plug in as specified in this project's README. I believe the compile error is coming from a **let** declaration in the webpack-dev-server source.  a straight webpack compile works. Removing uglification makes dev-server compile work.

**Comment by wangerniu** [comment]
the same appeared in my project, so how can we do to resolve it ?

**Comment by dylanseago** [comment]
@gabefinch I was using the same packages as you and getting an error on a **let** declaration as well. I was able to get it working by running webpack with:
`webpack --define process.env.NODE_ENV="production"` 
instead of 
`webpack -p`
Apparently, the -p flag enables the built-in v0.4.6 uglifyjs plugin which isn't compatible with es2015+ features like **let**.

**Comment by philippeluickx** [comment]
Uglify does not support es6, you have to get uglify-es. If you use uglifyjs-webpack-plugin as of v1, then all should be ok.

**Comment by michael-ciniawsky** [comment]
`uglifyjs-webpack-plugin =< v0.4.6` is pretty much locked and in maintenance until `v1.0.0` ships with `webpack v4.0.0`. Please consider updating the plugin to `>= v1.0.0` in the meantime or try to trace the bug down any critical **implementation** bug to where is occurs in the plugin and we maybe able to fix it in `v0.4.6`. Otherwise the best solution for any eventual  ES2015 related bug is to update to `v1.0.0`
